### DISCUSSION TITLE
Long term support for repo?

### DISCUSSION THREAD

**Discussion by kalegd** [protectwise/troika#287] [Q&A] [answered]
Hi, I'm thinking of migrating off of three-mesh-ui and on to troika-three-text. I just wanted to make sure before doing such a major overhaul of my project that the project is alive and well for the long run? I like to keep updating to the latest versions of three.js and don't want to get into a situation where I can't update for months at a time due to some breaking change on their side

Also just wanted to say this is a very cool project! Being able to drop pre-generated msdf files and supporting different character sets together out of the box is absolutely awesome

**Comment by lojjic** [comment, marked as answer]
Thanks for the question and the kind words! Yes, troika-three-text specifically will be maintained for the foreseeable future. The rest of the troika framework is less certain, and I imagine troika-three-text will be moved to a separate repo/org at some point to make the distinction clearer.

Keep in mind three-mesh-ui does things troika-three-text does not, like layouts and backgrounds. Ideally ttt would be integrated into three-mesh-ui at some point to get the best of both worlds.

**Reply by kalegd** [reply]
Great, thanks for letting me know! And yes, this will be a very painful migration, but one I think will be worth it in the end. Unfortunately there have been no code changes to three-mesh-ui in a while now :/
